### PR TITLE
feat: s3key info module

### PR DESCRIPTION
**Pull request by avirtopeanu-ionos**
## What does this fix or implement?

Added module to get all s3keys for a given user id. This helps users make more customized playbooks and also helps us with our tests (see #69: instead of checking if two idempotently-created s3keys are equal, perform a GET operation and check if only one was created)

## Checklist

- [x] PR name added as appropriate (e.g. feat/fix/doc/test/refactor/etc)
- [x] Documentation updated
- [x] Jira task updated
- [x] Wrote tests

### CODE REVIEW

File: plugins/modules/s3key_info.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+      s3key_info:
+        user_id: "{{ user_id }}"
+        register: s3key_info_response
+
+    - name: Show S3Keys
+      debug:
+        var: s3key_info_response.result
+'''
+
+def _get_request_id(headers):
```
Comment (Raduionutz) [inline plugins/modules/s3key_info.py:96]: I don't think this is used

File: plugins/modules/s3key_info.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    else:
+        raise Exception("Failed to extract request ID from response "
+                        "header 'location': '{location}'".format(location=headers['location']))
+
+
+def get_s3keys(module, client):
+    user_id = module.params.get('user_id')
+
+    user_s3keys_server = ionoscloud.UserS3KeysApi(client)
+    s3key_list = user_s3keys_server.um_users_s3keys_get(user_id=user_id, depth=5)
```
Comment (Raduionutz) [inline plugins/modules/s3key_info.py:109]: The response here does not seem to be used

File: plugins/modules/s3key_info.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+        try:
+            results = []
+
+            module.exit_json(**get_s3keys(module, api_client))
+        except Exception as e:
+            module.fail_json(msg='failed to set {object_name} state {state}: {error}'.format(object_name=OBJECT_NAME, error=to_native(e), state=state))
+
+
+if __name__ == '__main__':
+    main()
```
Comment (Raduionutz) [inline plugins/modules/s3key_info.py:195]: Could you add a newline at the end?

File: docs/api/user-management/s3key_info.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+```
+### Available parameters:
+&nbsp;
+
+| Name | Required | Type | Default | Description |
+| :--- | :---: | :--- | :--- | :--- |
+| user_id | True | str |  | The ID of the user |
+| api_url | True | str |  | The Ionos API base URL. |
+| username | True | str |  | The Ionos username. Overrides the IONOS_USERNAME environment variable. |
+| password | True | str |  | The Ionos password. Overrides the IONOS_PASSWORD environment variable. |
```
Comment (mihaela-mh) [inline docs/api/user-management/s3key_info.md:28]: Isn't **token** a parameter for this module too?

File: plugins/modules/s3key_info.py [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+            results.append(s3key.to_dict())
+        return {
+            'action': 'info',
+            'changed': False,
+            's3keys': results
+        }
+
+    except Exception as e:
+        module.fail_json(msg="failed to list the s3keys: %s" % to_native(e))
+        return {
```
Comment (mihaela-mh) [inline plugins/modules/s3key_info.py:120]: The **return** can be ommited. The program will stop the execution at **module.fail_json**

File: tests/s3key-info-test.yml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    - name: List S3Keys
+      s3key_info:
+        user_id: "{{ user_response.user.id }}"
+      register: s3keys_info_response
+
+    - name: Debug - Show response of s3keys_info
+      debug:
+        msg: "{{ s3keys_info_response.s3keys }}"
+
+    - name: Delete user
```
Comment (mihaela-mh) [inline tests/s3key-info-test.yml:41]: I suggest removing the s3keys before deleting the user. In this case, the deletion of the user will trigger deletion of the s3keys too, but in general there may be dependencies between resources and the cleanup part could fail.

**Review by mihaela-mh — APPROVED** [review]
Great work here! Congrats! 👍🏻

File: plugins/modules/s3key_info.py (hunk 5 above)
Comment (avirtopeanu-ionos) [reply plugins/modules/s3key_info.py:120]: thanks for noticing!

File: tests/s3key-info-test.yml (hunk 6 above)
Comment (avirtopeanu-ionos) [reply tests/s3key-info-test.yml:41]: that makes sense, thanks!

**Pull request merged**